test/test_pipeline/components/data_preprocessing/test_data_preprocessing_categorical.py

```diff
@@ -0,0 +1,71 @@
+import unittest
+import numpy as np
+from scipy import sparse
+
+from autosklearn.pipeline.components.data_preprocessing.data_preprocessing_categorical \
+    import CategoricalPreprocessingPipeline
+
+
+class CategoricalPreprocessingPipelineTest(unittest.TestCase):
+
+    def test_fit_transform(self):
+        X = np.array([
+            [1, 2, 0],
+            [3, 0, 0],
+            [2, 9, np.nan]])
+        Y = np.array([
+            [1, 0, 0, 0, 1, 0, 0, 1],
+            [0, 0, 1, 1, 0, 0, 0, 1],
+            [0, 1, 0, 0, 0, 1, 1, 0]])
+        # dense input
+        Yt = CategoricalPreprocessingPipeline().fit_transform(X)
+        self.assertTrue((Yt.todense() == Y).all())
+        # sparse input
+        X_sparse = sparse.csc_matrix(X+1)
+        Yt = CategoricalPreprocessingPipeline().fit_transform(X_sparse)
+        self.assertTrue((Yt.todense() == Y).all())
+
+    def test_transform(self):
+        X1 = np.array([
+            [1, 2, 0],
+            [3, 0, 0],
+            [2, 9, np.nan]])
+        Y1 = np.array([
+            [1, 0, 0, 0, 1, 0, 0, 1],
+            [0, 0, 1, 1, 0, 0, 0, 1],
+            [0, 1, 0, 0, 0, 1, 1, 0]])
+        X2 = np.array([
+            [2, 2, 1],
+            [3, 0, 0],
+            [2, np.nan, np.nan]])
+        Y2 = np.array([
+            [0, 1, 0, 0, 1, 0, 0, 0],
+            [0, 0, 1, 1, 0, 0, 0, 1],
+            [0, 1, 0, 0, 0, 0, 1, 0]])
+        # "fit"
+        CPPL = CategoricalPreprocessingPipeline()
+        CPPL.fit_transform(X1)
+        # Transform what was fitted
+        Y1t = CPPL.transform(X1)
+        self.assertTrue((Y1t.todense() == Y1).all())
+        # Transform a new dataset with categories not seen during fit
+        Y2t = CPPL.transform(X2)
+        self.assertTrue((Y2t.todense() == Y2).all())
+
+    def test_transform_with_coalescence(self):
```

Comment:
Could you please perform an extra `transform` here, too?

Reply:
Ok.

Reply:
I think you added this to the wrong test.

Reply:
Done. 
This time in the right test. I suppose.
:-)

Reply:
Yes, it is.